lib/Scheduler/sched_region.cpp

```diff
@@ -164,319 +156,302 @@ FUNC_RESULT SchedRegion::FindOptimalSchedule( //TODO: CHIPPIE: Add helper functi
-      //TODO: CHIPPIE: Determine what needs to be done between both? And determine what needs to be done in the case of heuristic schedule already having run?
-      bestSched = bestSched_ = initialSchedule = acoSchedule;
-      bestSchedLngth_ = initialScheduleLength_ = initialScheduleLength = ACOScheduleLength;
-      bestCost_ = initialScheduleCost_ = initialScheduleCost = ACOScheduleCost;
-      cout << "[CHIPPIE: DEBUG] ***** best length = " << bestSchedLngth_ << ", lower bound = " << schedLwrBound_ << ", ACO len =" << ACOScheduleLength << " *****\n"; //TODO: CHIPPIE: REMOVE DEBUGGING STATEMENT.
-      assert(bestSchedLngth_ >= schedLwrBound_);
-
-      initialScheduleOptimal = isACOOptimal;
-
-      //TODO: CHIPPIE: Done with this portion?
+    // B) Heuristic was never run. In that case, just use ACO and run with its
+    // results, into B&B.
+    else if (!HeuristicSchedulerEnabled) {
+      bestSched = bestSched_ = AcoSchedule;
+      bestSchedLngth_ = AcoScheduleLength_;
+      bestCost_ = AcoScheduleCost_;
+      // C) Neither scheduler was optimal. In that case, compare the two
+      // schedules and use the one that's better as the input (initialSched) for
+      // B&B.
     } else {
-      //TODO: CHIPPIE: Does anything else need to happen here? Nominally, the upper bound would need to be reset if ACO was not the lower one, but the compute upper bound code should already handle that sufficiently.
+      compareSchedules(lstSched, AcoSchedule, bestSched_);
+      bestSched = bestSched_;
+      bestSchedLngth_ = bestSched_->GetCrntLngth();
+      bestCost_ = bestSched_->GetCost();
     }
+  }
+  // Step #3: Compute the cost upper bound.
+  Milliseconds boundStart = Utilities::GetProcessorTime();
+  assert(bestSchedLngth_ >= schedLwrBound_);
+  assert(schedLwrBound_ <= bestSched_->GetCrntLngth());
+
+  // Calculate upper bounds with the best schedule found
+  CmputUprBounds_(bestSched_, false);
+  boundTime = Utilities::GetProcessorTime() - boundStart;
+  stats::boundComputationTime.Record(boundTime);
+
+#ifdef IS_DEBUG_PRINT_SCHEDS
+  lstSched->Print(Logger::GetLogStream(), "Heuristic");
+#endif
+#ifdef IS_DEBUG_PRINT_BOUNDS
+  dataDepGraph_->PrintLwrBounds(DIR_FRWRD, Logger::GetLogStream(),
+                                "CP Lower Bounds");
+#endif
 
-     Logger::Info(
-         "The ACO schedule is of length %d and spill cost %d. Tot cost = %d",
-         acoScheduleLength_, acoSchedule->GetSpillCost(), bestCost_);
-
-    //TODO: CHIPPIE: There's some other stuff that the old combined heuristic-aco scheduler code has...what of that needs to be copied, and how does it need to be modified?
-    //For example, the debug logs that were here originally, plus the #ifdef'd code chunks.
-  } else {
-    if (acoSchedulerEnabled) {
-      cout << "TODO: ACO Scheduler not run, but enabled (heuristic is optimal!).\n"; //TODO: CHIPPIE: Remove this when done debugging.
+  if (rgnTimeout == 0)
+    isLstOptml = true;
```

Comment:
So we're keeping this in? Will it be phased out in the future, as the BB disable flag makes it redundant?

Reply:
I believe this is still useful for disabling a pass on the two-pass scheduler but I agree we will need to phase it out to an option in sched.ini. I updated it to set "BbSchedulerEnabled" to false instead of isLstOptml to true for clarity.

Reply:
I believe this is still useful for disabling a pass on the two-pass scheduler but I agree we will need to phase it out to an option in sched.ini. I updated it to set "BbSchedulerEnabled" to false instead of isLstOptml to true for clarity.

Reply:
Got it. Sounds good.